packages/js-commons/src/local-files.ts

```diff
@@ -0,0 +1,262 @@
+  get: async (appId: string): Promise<ApplicationDTO | null> => {
+    const userAppFolder = join(hiddenEnsembleFolder, appId);
+    if (!existsSync(userAppFolder)) return null;
+
+    const [
+      env,
+      theme,
+      assets,
+      screens,
+      widgets,
+      scripts,
+      groupLabels,
+      translations,
+    ] = await Promise.all([
+      readJsonFile<EnvironmentDTO>(join(userAppFolder, ArtifactFilesName.env)),
+      readJsonFile<ThemeDTO>(join(userAppFolder, ArtifactFilesName.theme)),
+      readJsonFile<AssetDTO[]>(join(userAppFolder, ArtifactFilesName.assets)),
+      readJsonFile<ScreenDTO[]>(join(userAppFolder, ArtifactFilesName.screens)),
+      readJsonFile<WidgetDTO[]>(join(userAppFolder, ArtifactFilesName.widgets)),
+      readJsonFile<ScriptDTO[]>(join(userAppFolder, ArtifactFilesName.scripts)),
+      readJsonFile<Record<string, string>>(
+        join(userAppFolder, ArtifactFilesName.groupLabels),
+      ),
+      readJsonFile<LanguageDTO[]>(
+        join(userAppFolder, ArtifactFilesName.translations),
+      ),
+    ]);
+
+    const applicationData: ApplicationDTO = {
+      env,
+      theme,
+      assets,
+      widgets,
+      scripts,
+      name: "",
+      id: appId,
+      translations,
+      screens: screens ?? [],
+      groupLabels: convertJsonToMap(groupLabels ?? {}),
+    };
+    return applicationData;
+  },
+  put: async (appData: ApplicationDTO): Promise<void> => {
+    const userAppFolder = join(hiddenEnsembleFolder, appData.id);
+    ensureDir(userAppFolder);
+
+    const jsonFilesPath = {
+      env: join(userAppFolder, ArtifactFilesName.env),
+      theme: join(userAppFolder, ArtifactFilesName.theme),
+      assets: join(userAppFolder, ArtifactFilesName.assets),
+      screens: join(userAppFolder, ArtifactFilesName.screens),
+      widgets: join(userAppFolder, ArtifactFilesName.widgets),
+      scripts: join(userAppFolder, ArtifactFilesName.scripts),
+      groupLabels: join(userAppFolder, ArtifactFilesName.groupLabels),
+      translations: join(userAppFolder, ArtifactFilesName.translations),
+    };
+
+    await Promise.all([
+      writeJsonData(jsonFilesPath.env, appData.env ?? {}),
+      writeJsonData(jsonFilesPath.theme, appData.theme ?? {}),
+      writeJsonData(jsonFilesPath.assets, appData.assets ?? []),
+      writeJsonData(jsonFilesPath.screens, appData.screens ?? []),
+      writeJsonData(jsonFilesPath.scripts, appData.scripts ?? []),
+      writeJsonData(jsonFilesPath.widgets, appData.widgets ?? []),
+      writeJsonData(jsonFilesPath.groupLabels, appData.groupLabels ?? []),
+      writeJsonData(jsonFilesPath.translations, appData.translations ?? []),
+    ]);
+  },
+});
+
+export const getYamlApplicationTransporter = (
+  yamlFolder: string,
+  globalFolder: string,
+): YamlApplicationTransporter => ({
+  get: async (
+    appId: string,
+    documentId: string,
+    documentType: string,
+  ): Promise<string> => {
+    const appsMetaData = await getAppMetaData(globalFolder);
```

Comment:
App metadata should live next to YAML files.

Reply:
This is for the Global Apps MetaData as we discussed. A global single file that holds generic metadata of each app and location where the YAML and JSON files of each app is located.